test/steps/dockerfile_provider.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package steps
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// KogitoApplicationDockerfileProvider is the API to provide Dockerfile content for image creation based on built project content
+type KogitoApplicationDockerfileProvider interface {
+	// GetDockerfileContent returns Dockerfile content for image creation
+	GetDockerfileContent() string
+}
+
+type kogitoApplicationDockerfileProviderStruct struct {
+	projectLocation         string
+	baseImageName           string
+	applicationBinarySuffix string
+	libFolderNeeded         bool
+}
+
+var quarkusNonNativeKogitoApplicationDockerfileProvider = kogitoApplicationDockerfileProviderStruct{
+	baseImageName:           "quay.io/kiegroup/kogito-quarkus-jvm-ubi8",
```

Comment:
The base image names are already in kogito_imagestream.go, concretely, we might make use of them by:
BuildImageStreams[BuildTypeRuntime][SpringbootRuntimeType] // for spring
BuildImageStreams[BuildTypeRuntime][QuarkusRuntimeType] // for quarkus native
BuildImageStreams[BuildTypeRuntimeJvm][QuarkusRuntimeType] // for quarkus non native

Reply:
Good catch, will update it.